pkg/ignition/ignition.go

```diff
@@ -0,0 +1,120 @@
+package ignition
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	ignCfg "github.com/coreos/ignition/v2/config/v3_2"
+	ignCfgTypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcfg "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups="machineconfiguration.openshift.io",resources=machineconfigs,verbs=list;watch
+
+const (
+	// kubeletSystemdName is the name of the systemd service that the kubelet runs under,
+	// this is used to parse the kubelet args
+	kubeletSystemdName = "kubelet.service"
+	// CloudConfigOption is the kubelet CLI option for the cloud configuration file
+	CloudConfigOption = "cloud-config"
+	// CloudProviderOption is the kubelet CLI option for cloud provider
+	CloudProviderOption = "cloud-provider"
+	// RenderedWorkerPrefix allows identification of the rendered worker MachineConfig, the combination of all worker
+	// MachineConfigs.
+	RenderedWorkerPrefix = "rendered-worker-"
+)
+
+// GetKubeletArgs returns a set of arguments for kubelet.exe, as specified in the ignition file
+func GetKubeletArgs(c client.Client) (map[string]string, error) {
+	log := ctrl.Log.WithName("ignition")
+
+	machineConfigs := &mcfg.MachineConfigList{}
+	err := c.List(context.TODO(), machineConfigs)
+	if err != nil {
+		return nil, err
+	}
+	// Grab the latest rendered-worker MachineConfig by sorting the MachineConfig list by the latest creation
+	// timestamp first.
+	sort.Slice(machineConfigs.Items, func(i, j int) bool {
+		iTimestamp := machineConfigs.Items[i].GetCreationTimestamp()
+		jTimestamp := machineConfigs.Items[j].GetCreationTimestamp()
+		return jTimestamp.Before(&iTimestamp)
+	})
+	var rawIgnition []byte
+	for _, mc := range machineConfigs.Items {
+		if strings.HasPrefix(mc.Name, RenderedWorkerPrefix) {
+			rawIgnition = mc.Spec.Config.Raw
+			log.Info("Received ignition spec", "MachineConfig", mc.Name)
```

Comment:
I'm not sure the value of this considering this would only be used when bootstrapping, its a lot more valuable for linux nodes. I've added [another story](https://issues.redhat.com/browse/WINC-897) in a TODO which is semi-related.

Reply:
The value would be when debugging a customer issue. We can easily look at the node object and figure out where we got the kubelet args from.